File: dmn-test-cases/standard/other/1.4/decision-table-with-annotations/translator/expected/java/dmn/type/TAImpl.java
package type;

import java.util.*;

@javax.annotation.Generated(value = {"itemDefinition.ftl", "tA"})
@com.fasterxml.jackson.annotation.JsonPropertyOrder(alphabetic = true)
public class TAImpl implements TA {
        private String name;
        private java.math.BigDecimal price;

    public TAImpl() {
    }

    public TAImpl(String name, java.math.BigDecimal price) {
        this.setName(name);
        this.setPrice(price);
    }

    @com.fasterxml.jackson.annotation.JsonGetter("name")
    public String getName() {
        return this.name;
    }

    @com.fasterxml.jackson.annotation.JsonSetter("name")
    public void setName(String name) {
        this.name = name;
    }

    @com.fasterxml.jackson.annotation.JsonGetter("price")
    public java.math.BigDecimal getPrice() {
        return this.price;
    }

    @com.fasterxml.jackson.annotation.JsonSetter("price")
    public void setPrice(java.math.BigDecimal price) {
        this.price = price;
    }

    @Override
    public boolean equals(Object o) {
        return equalTo(o);
    }

    @Override
    public int hashCode() {
        return hash();
    }

    @Override
    public String toString() {
        return asString();
    }
}

File: dmn-core/src/test/resources/tck/cl3/expected/kotlin/dmn/0017-tabletests/type/TAImpl.kt
package type

import java.util.*

@javax.annotation.Generated(value = ["itemDefinition.ftl", "tA"])
@com.fasterxml.jackson.annotation.JsonPropertyOrder(alphabetic = true)
class TAImpl : TA {
    @get:com.fasterxml.jackson.annotation.JsonGetter("name")
    @set:com.fasterxml.jackson.annotation.JsonGetter("name")
    override var name: String? = null

    @get:com.fasterxml.jackson.annotation.JsonGetter("price")
    @set:com.fasterxml.jackson.annotation.JsonGetter("price")
    override var price: java.math.BigDecimal? = null

    constructor() {
    }

    constructor (name: String?, price: java.math.BigDecimal?) {
        this.name = name
        this.price = price
    }

    override fun equals(other: Any?): Boolean {
        return equalTo(other)
    }

    override fun hashCode(): Int {
        return hash()
    }

    @Override
    override fun toString(): String {
        return asString()
    }
}
